### DISCUSSION TITLE
Job environment can't access services

### DISCUSSION THREAD

**Discussion by zaherg** [community/community#25603] [Actions]
<p>I was happy to know that the global environment is now alive and can be within the job or the workflow (<a href="https://github.blog/changelog/2019-10-01-github-actions-new-workflow-syntax-features/#env-at-the-workflow-and-job-level" rel="nofollow noopener">https://github.blog/changelog/2019-10-01-github-actions-new-workflow-syntax-features/#env-at-the-workflow-and-job-level</a>)</p>
<p>but for some reason, the env at the job level can’t access any information about the services, for example, something like the following</p>
<pre><code>name: CI
on: [push, pull_request]
jobs:
  phpunit:
    name: PHPUnit
    runs-on: ubuntu-latest
    env:
      DB_PORT: ${{ job.services.mysql.ports['3306'] }}
      DB_PASSWORD: password
      BROADCAST_DRIVER: log
      CACHE_DRIVER: redis
      QUEUE_CONNECTION: redis
      SESSION_DRIVER: redis
    services:
      mysql:
        image: mysql:5.7
        env:
          MYSQL_ALLOW_EMPTY_PASSWORD: false
          MYSQL_ROOT_PASSWORD: password
          MYSQL_DATABASE: laravel
        ports:
          - 3306
        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
</code></pre>
<p>Will not work, neither</p>
<pre><code>jobs:
  phpunit:
    name: PHPUnit
    runs-on: ubuntu-latest
    env:
      DB_PORT: ${{ services.mysql.ports['3306'] }}
      DB_PASSWORD: password
      BROADCAST_DRIVER: log
      CACHE_DRIVER: redis
      QUEUE_CONNECTION: redis
      SESSION_DRIVER: redis
</code></pre>
<p>The error I get back:</p>
<p><em>Your workflow file was invalid: The pipeline is not valid. .github/workflows/ci.yml (Line: 26, Col: 16): Unrecognized named-value: ‘job’. Located at position 1 within expression: job.services.mysql.ports[‘3306’]</em></p>
<p>Any idea how to fix that, other than moving the env to the task that need to connect to the database? or it is by design?</p>

**Comment by chrispat** [comment, marked as answer]
<p>That is currently by design.  The job level ENV is built before the services are setup on the runner so those values are not known.   Your only option is to pass the variable to the step where it is needed.</p>

**Comment by zaherg** [comment]
<p>hey <a class="mention" href="/chrispat">@chrispat</a>  is there any plan to change this behavior?</p>

**Comment by chrispat** [comment]
<p>Not at the moment.  We will continue to evaluate based on feedback.</p>

**Comment by zaherg** [comment]
<p>Thank you very much :+1:</p>

**Comment by chingc** [comment]
<p><span class="mention">@chrispatI</span>’ve also run into this issue and would love to be able to do what <a class="mention" href="/linuxjuggler">@linuxjuggler</a> suggests.  I have multiple steps that require the database port value.</p>
